platform/pocketbook/koreader.app

```diff
@@ -76,11 +78,108 @@ if [ -e crash.log ]; then
 fi
 
 export KO_EXIT_CODE="/tmp/.koreader.exit"
-RETURN_VALUE="85"
-while [ "${RETURN_VALUE}" = "85" ]; do
-    rm -f "${KO_EXIT_CODE}"
+CRASH_COUNT=0
+CRASH_TS=0
+CRASH_PREV_TS=0
+# List of supported special return codes
+KO_RC_RESTART=85
+#KO_RC_USBMS=86
+# Ensure a clean slate on startup
+rm -f "${KO_EXIT_CODE}"
+RETURN_VALUE=${KO_RC_RESTART}
+while [ ${RETURN_VALUE} -ne 0 ]; do
+    if [ ${RETURN_VALUE} -eq ${KO_RC_RESTART} ]; then
+        # Do an update check now, so we can actually update KOReader via the "Restart KOReader" menu entry ;).
+        ko_update_check
+    fi
+
     ./reader.lua "${args}" >>crash.log 2>&1
-    RETURN_VALUE=$(cat ${KO_EXIT_CODE})
+    RETURN_VALUE="$(cat ${KO_EXIT_CODE})"
+    rm -f "${KO_EXIT_CODE}"
+
+    # Did we crash?
+    if [ ${RETURN_VALUE} -ne 0 ] && [ ${RETURN_VALUE} -ne ${KO_RC_RESTART} ]; then
```

Comment:
On  crash, the file in ${KO_EXIT_CODE} will be always empty, so perhaps one could just do `if [ -e ${KO_EXIT_CODE} ]`. While these conditions will hold true in the empty case, it's more of a side-effect fluke than stating what really goes on explicitly.

Reply:
Yeah, good point ;).